### DISCUSSION TITLE
NWDAF APIs and exposure for the AmariSoft collector

### DISCUSSION THREAD

**Discussion by xilouris** [core-ncsrd/.github#3] [Ideas]
# Team Members 

- @mchristopoulou 
- @xilouris 
- + developer (TBD)

# Objective 

Develop the function and the APIs that will allow Amarisoft collected data to be available to other internal or external NFs of the 5G Core.

# Definitions

NWDAF is designed to overcome market fragmentation and proprietary solutions in the area of network analytics, streamlining the way core network data is produced and consumed, as well as generating insights and taking actions based on these insights.

NWDAF addresses three primary standardization points:

- Data collection interface from network nodes
- Predefined analytics insights
- Data exposure interface for consumers 

# Resources 

From the 3GPP:

- [TS 29520 - Network Data Analytics Services](https://www.etsi.org/deliver/etsi_ts/129500_129599/129520/16.04.00_60/ts_129520v160400p.pdf) 
Table 5.1.6.1-1 specifies the data types defined for the Nnwdaf_EventsSubscription service based interface protocol.

- [TS29520_Nnwdaf_AnalyticsInfo.yaml](https://www.3gpp.org/ftp/Specs/archive/OpenAPI/Rel-16/TS29520_Nnwdaf_AnalyticsInfo.yaml)
- [TS29520_Nnwdaf_EventsSubscription.yaml](https://www.3gpp.org/ftp/Specs/archive/OpenAPI/Rel-16/TS29520_Nnwdaf_EventsSubscription.yaml)

**Comment by mchristopoulou** [comment]
From the 3GPP:

[TS 29520 - Network Data Analytics Services](https://www.etsi.org/deliver/etsi_ts/129500_129599/129520/16.04.00_60/ts_129520v160400p.pdf) 

Table 5.1.6.1-1 specifies the data types defined for the Nnwdaf_EventsSubscription service based interface protocol.

[TS29520_Nnwdaf_AnalyticsInfo.yaml](https://www.3gpp.org/ftp/Specs/archive/OpenAPI/Rel-16/TS29520_Nnwdaf_AnalyticsInfo.yaml)

[TS29520_Nnwdaf_EventsSubscription.yaml](https://www.3gpp.org/ftp/Specs/archive/OpenAPI/Rel-16/TS29520_Nnwdaf_EventsSubscription.yaml)

**Reply by xilouris** [reply]
included in the description of the discussion
